### ISSUE TITLE
[TW-1145] time_t/int collision on 32-bit system

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1180]
_Paul Beckingham on 2009-07-23T17:21:57Z says:_ 

 Added by Pietro Cerutti about 7 hours ago

On 32-bit platforms, time_t is often defined as int.

class Duration defines two overloaded operators,

43   operator int ();

44   operator time_t ();
which are implemented as

54 ////////////////////////////////////////////////////////////////////////////////

55 Duration::operator int ()

56 {

57   return (int) mDays;

58 }

59 

60 ////////////////////////////////////////////////////////////////////////////////

61 Duration::operator time_t ()

62 {

63   return mDays;

64 }
if time_t is #define'd as int, the compilation fails:

c++ -DHAVE_CONFIG_H -I. -I.. -O2 -fno-strict-aliasing -pipe -Wall -pedantic -O3 -MT Duration.o -MD -MP -MF .deps/Duration.Tpo -c -o Duration.o Duration.cpp
Duration.cpp:61: error: redefinition of 'Duration::operator time_t()'
Duration.cpp:55: error: 'Duration::operator int()' previously defined here
Error code 1

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2009-07-23T17:21:57Z
Modified: 2014-02-09T02:37:22Z
```

**Comment by taskwarrior** [comment]
_Pietro Cerutti on 2009-07-24T03:22:41Z says:_ 

 I suggest to remove the int() operator at all, and cast to an integer type in the caller, when required, e.g., for printf.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2009-07-26T16:18:21Z says:_ 

 Fixed in task 1.8.1.  Patch offered by Pietro Cerutti applied.

Although this patch was applied to the task 1.8.1 branch, it may never be released as 1.8.1.  The reasoning behind this decision is:

a) Pietro, although affected by this problem, fixed it for himself.  No one else has reported this, so this is not a critical fix unless someone else reports it.

b) It is likely that other, more critical bugs are found, in which case 1.8.1 will be released, and will include this fix.

c) In the event that no one else reports this problem, and no other critical bugs are found in 1.8.0, the 1.8.1 branch will be merged to 2.0.0 and released there.

**Comment by pbeckingham** [comment]
[TW-1145_patch-Duration.txt](https://github.com/GothenburgBitFactory/taskwarrior/files/1756300/TW-1145_patch-Duration.txt)
